### PR TITLE
Make JS/TS Function templates async by default

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :broom: Make JS/TS Function template Functions async by default.
  Rationale: `async` is almost necessary for any non-trivial functionality.

### CODE REVIEW

File: templates/node/cloudevents/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const { CloudEvent, HTTP } = require('cloudevents');
  * See: https://github.com/knative-sandbox/kn-plugin-func/blob/main/docs/guides/nodejs.md#the-context-object
  * @param {CloudEvent} event the CloudEvent
  */
-function handle(context, event) {
+const handle = async (context, event) => {
```
Comment (lance) [inline templates/node/cloudevents/index.js:21]: Why not just add the `async` keyword as in `async function handle()`?

Comment (matejvasek) [reply templates/node/cloudevents/index.js:21]: No particular reason, I just like arrow functions.

Comment (matejvasek) [reply templates/node/cloudevents/index.js:21]: Also `function` binds `this` to something. Do we want that something expose?

Comment (zroubalik) [reply templates/node/cloudevents/index.js:21]: I am not node expert, but I tend to agree with Lance, it is more obvious for me that way. But again, I am not expert in this area, so I am fine with any approach.

Comment (matejvasek) [reply templates/node/cloudevents/index.js:21]: @lance are we OK with arrow functions or shall I revert to `function`.

**Comment by matejvasek** [comment]
@lance PTAL

**Pull request merged**